server-common/src/main/java/org/apache/kafka/server/group/share/SharePartitionKey.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.group.share;
+
+import org.apache.kafka.common.Uuid;
+
+import java.util.Objects;
+
+/**
+ * Common immutable share partition key class. This class is
+ * placed in server-common so that it can be freely used across
+ * various modules.
+ */
+public class SharePartitionKey {
+    private final String groupId;
+    private final Uuid topicId;
+    private final int partition;
+
+    public SharePartitionKey(String groupId, Uuid topicId, int partition) {
+        this.groupId = groupId;
+        this.topicId = topicId;
+        this.partition = partition;
+    }
+
+    public static class Builder {
+        private String groupId;
+        private Uuid topicId;
+        private int partition;
+
+        public Builder setGroupId(String groupId) {
+            this.groupId = groupId;
+            return this;
+        }
+
+        public Builder setTopicId(Uuid topicId) {
+            this.topicId = topicId;
+            return this;
+        }
+
+        public Builder setPartition(int partition) {
+            this.partition = partition;
+            return this;
+        }
+
+        public SharePartitionKey build() {
+            return new SharePartitionKey(groupId, topicId, partition);
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof SharePartitionKey)) return false;
+        SharePartitionKey that = (SharePartitionKey) o;
+        return partition == that.partition && Objects.equals(groupId, that.groupId) && Objects.equals(topicId, that.topicId);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(groupId, topicId, partition);
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%s:%s:%d", groupId, topicId, partition);
```

Comment:
Is there a reason we don't want the usual "toString" format here?

Reply:
We need share part keys `(groupId:topicId:partition)` as strings as well because that is mandated by some APIs like the FindCoordinator requests. 
Keeping this format will allow to use `SharePartitionKey` interchangeably.

Reply:
Ok, in that case lets create a separate method for the specific format we need.

Reply:
committed in the subsequent revision, plz check once again